nats-compiler-plugin/src/main/java/io/ballerina/stdlib/nats/plugin/NatsFunctionValidator.java

```diff
@@ -0,0 +1,304 @@
+        RequiredParameterNode requiredParameterNode = (RequiredParameterNode) parameterNode;
+        Node parameterTypeNode = requiredParameterNode.typeName();
+        SemanticModel semanticModel = context.semanticModel();
+        Optional<Symbol> paramSymbol = semanticModel.symbol(parameterTypeNode);
+        if (paramSymbol.isPresent()) {
+            Optional<ModuleSymbol> moduleSymbol = paramSymbol.get().getModule();
+            if (moduleSymbol.isPresent()) {
+                String orgName = moduleSymbol.get().id().orgName();
+                String moduleName = moduleSymbol.get().id().moduleName();
+                String paramName = paramSymbol.get().getName().isPresent() ?
+                        paramSymbol.get().getName().get() : "";
+                if (!moduleName.equals(PluginConstants.PACKAGE_PREFIX) ||
+                        !orgName.equals(PluginConstants.PACKAGE_ORG)) {
+                    context.reportDiagnostic(PluginUtils.getDiagnostic(
+                            CompilationErrors.INVALID_FUNCTION_PARAM_MESSAGE,
+                            DiagnosticSeverity.ERROR, requiredParameterNode.location()));
+                } else {
+                    if (!paramName.equals(PluginConstants.MESSAGE)) {
+                        String typeName = getTypeDefinitionNameForMessage();
+                        if (!paramName.equals(typeName)) {
+                            context.reportDiagnostic(PluginUtils.getDiagnostic(
+                                    CompilationErrors.INVALID_FUNCTION_PARAM_MESSAGE,
+                                    DiagnosticSeverity.ERROR, requiredParameterNode.location()));
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private String getTypeDefinitionNameForMessage() {
+        SemanticModel semanticModel = context.semanticModel();
+        List<Symbol> moduleSymbols = semanticModel.moduleSymbols();
+        for (Symbol symbol : moduleSymbols) {
+            if (symbol.kind() == SymbolKind.TYPE_DEFINITION) {
+                TypeDefinitionSymbol definitionSymbol = (TypeDefinitionSymbol) symbol;
+                if (definitionSymbol.typeDescriptor().typeKind() == TypeDescKind.RECORD) {
+                    Map<String, RecordFieldSymbol> record =
+                            ((RecordTypeSymbol) definitionSymbol.typeDescriptor()).fieldDescriptors();
+                    if (record.size() == 3 &&
+                            record.containsKey(PluginConstants.CONTENT_FIELD) &&
+                            record.containsKey(PluginConstants.SUBJECT_FIELD) &&
+                            record.containsKey(PluginConstants.REPLY_TO_FIELD)) {
+                        return definitionSymbol.getName().get();
+                    }
+                }
+            }
+        }
+        return null;
+    }
+
+    private void validateReturnTypeErrorOrNil(FunctionDefinitionNode functionDefinitionNode) {
```

Comment:
What's the exact requirement we have for the return type? I think this can be written in a simpler/cleaner way. Same goes for the other similar methods as well.

Reply:
it should be either `error?` of `nats:Error?`

Reply:
Ah I see. Then this is also fine I guess. But isn't this only checking for `error?` 🤔 and also for just `()`?

Reply:
if it's a union then only `error?/nats:Error?` is allowed. if it's not only `()` is allowed. There can't be anything like `string`, `int` etc.